pkg/nrtupdater/fake/fake.go

```diff
@@ -0,0 +1,106 @@
+package fake
+
+import (
+	"math/rand"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/klog/v2"
+
+	"github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha1"
+	"github.com/k8stopologyawareschedwg/resource-topology-exporter/pkg/dump"
+	"github.com/k8stopologyawareschedwg/resource-topology-exporter/pkg/nrtupdater"
+)
+
+type Generator struct {
+	Infos    <-chan nrtupdater.MonitorInfo
+	infoChan chan nrtupdater.MonitorInfo
+	interval time.Duration
+}
+
+func NewGenerator(interval time.Duration) *Generator {
+	gen := Generator{
+		infoChan: make(chan nrtupdater.MonitorInfo),
+		interval: interval,
+	}
+	gen.Infos = gen.infoChan
+	return &gen
+}
+
+func (ge *Generator) Run() {
+	ticker := time.NewTicker(ge.interval)
+	anns := make(map[string]string)
+
+	for {
+		select {
+		case <-ticker.C:
+			mi := nrtupdater.MonitorInfo{
+				Timer:       true,
+				Zones:       Zones(),
+				Annotations: anns,
```

Comment:
I don't see it's being used somewhere, why do we need to initialize that?

Reply:
good point, started more ambitious and then downsized. We don't need annotations yet.